docs/exit_validation.md

```diff
@@ -1,90 +1,94 @@
-    - trying to preempt underfunded chain situation, which would jeopardize user's funds
-2. let the user know about byzantine child chain as soon as is reasonably possible
-4. (optional) protect against loss in case of an invalid exit finalization
-5. prevent spurious prompts to mass exit that might result from root chain reorgs, race conditions etc.
-
-### Actions that the Watcher should prompt:
-1. If an exit is known to be invalid it should be challenged (almost) immediately
-2. If an exit is invalidated with a transaction submitted *before* `scheduled_finalization_time - sla_margin` it must be challenged
-3. If an exit is invalidated with a transaction submitted *after* `scheduled_finalization_time - sla_margin` it must be challenged **AND** watcher must prompt an exit **AND** watcher must not allow spending and depositing
-    - because our child chain implementation will never get close to this happening, so it is a symptom of a (subtle) hack attempt on the child chain server.
-    By "subtle" we mean that it is a hack that counts on getting away with exit invalidation being submitted too late, rather than on just dropping a huge invalid UTXO and attempting to exit from that
-4. If an exit is invalid and remains unchallenged within a short period of time (`sla_margin`) from its finalization, it must be challenged **AND** watcher must not allow spending and depositing, because:
-    a. we are running too close to the minimum safety requirements
-    b. Unchallenged exit creates a partial reserve situation and causes funds loss for some of users. There will be a loss of funds and it is not related to time.
-    This can happen when an exit, a spend that invalidates that exit and an exit of that invalidating spend all coincide and are unchallenged.
-5. (optional) If finalization of an invalid exit occurs, the watcher must prompt an exit without guarantees of recovery of funds
-
-Conditions 3 and 4 can be represented jointly, by **continuous** (every child block) validation of the exits.
-If any exit is invalid and its finalization is near, then actions listed under 3 and 4 should take place.
-Let's call this joint representation of 3 and 4 `unchallenged_exit`.
-More on this in the [`unchallenged_exit` condition](#unchallenged-exit-condition) section.
-
-All that takes as an assumption that:
+    - the Child Chain Server listens to every `ExitStarted` Root Chain event and _immediately_ "spends" the exited utxo.
+    - the Child Chain Server listens to every `InFlightExitStarted` Root Chain event and _immediately_ "spends" the exiting tx's **inputs**
+    - the Child Chain Server listens to every `InFlightExitPiggybacked` Root Chain event (on outputs) and _immediately_ "spends" the piggybacked outputs - as long as the IFEing tx has been included in the chain and the output exists
+  These rules prohibit the user from spending an UTXO exiting this way or another, preventing exit invalidation.
```

Comment:
This sentence needs work. Rules don't prohibit user, they prevent user from exploiting situation. "This way of another" is very hand-wavy. Maybe "rules prevent the user from spending on plasma chain an currently exiting UTXO, preventing exit invalidation"?

Reply:
Ha, nice catch^2, I put in 3x "prohibit" after Kevin's comment, but here it clearly doesn't work :joy: . I'll work on this :pray: